fontforge/sfd.c

```diff
@@ -1067,6 +1079,31 @@ static void SFDDumpImage(FILE *sfd,ImageList *img) {
     fprintf(sfd,"\nEndImage\n" );
 }
 
+#ifndef _NO_LIBPNG
+static void SFDDumpImagePNG(FILE *sfd,ImageList *img) {
+    struct enc85 enc = {0};
+    char* pngbuf;
+    size_t pnglen, i;
+
+    if (!GImageWritePngBuf(img->image, &pngbuf, &pnglen, 1, false)) {
+        IError("Failed to serialise PNG image");
+        return;
+    }
+
+    fprintf(sfd, "Image2: image/png %d %g %g %g %g\n",
+        (int)pnglen, (double) img->xoff, (double) img->yoff, (double) img->xscale, (double) img->yscale );
+
+    enc.sfd = sfd;
+    for (i = 0; i<pnglen; ++i) {
+        SFDEnc85(&enc, pngbuf[i]);
+    }
+    free(pngbuf);
```

Comment:
Ugh, you're going to hate me for this ...

So we're copying the buffer in `GImageWritePngBuf` so as not to "pollute g_free across function calls", the latter being a good thing. But even so we are assuming the buffer is allocated in a certain way and not with (for example) `chunkalloc()`. Why not just add and call a matching `GImageFreePngBuf`? No copy and the alloc/free can be right next to each other in the same file. 

Seems like better practice overall.

Reply:
@jtanx wrote that code, I don't really understand how to do what you're asking so will wait for him to answer and/or stick a commit on top of `images` or `PNG_in_SFD`

Reply:
I'll give it a shot.